models/data_model.go

```diff
@@ -81,6 +92,21 @@ type Table struct {
 	LinksToSingle map[LinkName]LinkToSingle
 }
 
+func (t Table) Copy() Table {
+	fields := make(map[FieldName]Field)
+	for k, v := range t.Fields {
+		fields[k] = v
+	}
+	links := make(map[LinkName]LinkToSingle)
+	for k, v := range t.LinksToSingle {
+		links[k] = v
+	}
+	out := t
+	out.Fields = fields
+	out.LinksToSingle = links
+	return out
+}
+
```

Comment:
IMO, we should be "dumb" on those :
- rename `Copy()` to `DeepCopy()` (so it's clear we do not return a shallow copy)
- change this perticular function impelmentation to return a deep copy (fields + links)

By experience, using deep copy protect us against weird bug (for now, we avoid this kind of bug inside the integration using a reasign statement, instead of a mutation)

Reply:
It's true that I did get a weird bug in my unit tests (didn't really reach a usecase error though) which is how I worked out that I need to have a deeper copy.
I'm not sure to understand what you mean by `change this perticular function impelmentation to return a deep copy (fields + links)` however, returning a deep copy if links & fields is precisely what I'm doing here